diff --git a/odb/qt/date-time/sqlite/qdate-time-traits.hxx b/odb/qt/date-time/sqlite/qdate-time-traits.hxx
index c62d513..c394bd4 100644
--- a/odb/qt/date-time/sqlite/qdate-time-traits.hxx
+++ b/odb/qt/date-time/sqlite/qdate-time-traits.hxx
@@ -57,7 +57,7 @@ namespace odb
           is_null = false;
 
           std::string s (
-            v.toString ("yyyy-MM-ddTHH:mm:ss.zzz").toStdString ());
+            v.toString ("yyyy-MM-ddTHH:mm:ss.zzz").toUtf8 ().constData ());
 
           n = s.size ();
           if (n > i.capacity ())
diff --git a/odb/qt/date-time/sqlite/qdate-traits.hxx b/odb/qt/date-time/sqlite/qdate-traits.hxx
index 80879a7..3303cea 100644
--- a/odb/qt/date-time/sqlite/qdate-traits.hxx
+++ b/odb/qt/date-time/sqlite/qdate-traits.hxx
@@ -57,7 +57,7 @@ namespace odb
         {
           is_null = false;
 
-          std::string s (v.toString ("yyyy-MM-dd").toStdString ());
+          std::string s (v.toString ("yyyy-MM-dd").toUtf8 ().constData ());
 
           n = s.size ();
           if (n > i.capacity ())
diff --git a/odb/qt/date-time/sqlite/qtime-traits.hxx b/odb/qt/date-time/sqlite/qtime-traits.hxx
index 391df35..fe4ebb5 100644
--- a/odb/qt/date-time/sqlite/qtime-traits.hxx
+++ b/odb/qt/date-time/sqlite/qtime-traits.hxx
@@ -57,7 +57,7 @@ namespace odb
         {
           is_null = false;
 
-          std::string s (v.toString ("HH:mm:ss.zzz").toStdString ());
+          std::string s (v.toString ("HH:mm:ss.zzz").toUtf8 ().constData ());
 
           n = s.size ();
           if (n > i.capacity ())